### ISSUE TITLE
Folding for markdown headings

### ISSUE THREAD

**Issue by jtkiley** [atom/language-gfm#20]
It would be nice to have markdown headings fold like the unordered and ordered lists currently do.

For comparison, TextMate folds the area between any heading and the next heading, regardless of level. Instead, I think it would be nicer if headings folded the region from any heading to just above the next heading of equal or higher level. That seems more logical, it follows the logic of dedicated GUI outliners, and it's friendlier to the obsessively organized.

I spent a few minutes looking though the package to see how it's done (in the hope of making a full request), but I never found it. I have basically zero experience with bundles, so that's not surprising, though I did find folding in TextMate's markdown bundle.

**Comment by KnutHelland** [comment]
I really miss this feature too and would like to contribute if I knew where to start. But it looks like it is quite a big job to implement. Atom folds by indentation level, and I don't know where to start and if it's even possible to do this in a package without touching core functionality in Atom.

But since it is also possible to fold selections, it should be possible to create custom hacky commands for folding markdown. But I would really prefer to use the default folding commands that I'm used to from other source code.

**Comment by melke** [comment]
Hi @KnutHelland , I just wrote a package that folds/unfolds markdown headings. It has it's own commands, so it can't hook into the normal folding commands, but at least it does the job. If you like, please try it out and send me feedback or PR:s with suggestions for improvements. https://github.com/melke/markdown-folder

**Comment by mm-gmbd** [comment]
@melke, thanks for putting this package together! I've just started working with a large markdown file, so this was perfect timing. I've submitted a couple issues/feature requests that may be of interest to you. I'd like to fork and PR, but I simply don't have the time as of now.

**Comment by KnutHelland** [comment]
I agree, your solution looks interesting @melke. Haven't had time to test it yet, but I looked at the code now and it looks very nice! Thank you for sharing!

**Comment by ChristopherKing42** [comment]
:+1:

**Comment by olee** [comment]
No activity on this?
It's really a bummer, that we cannot fold sections...

**Comment by ricpelo** [comment]
Any news?
